src/metabase/embed/settings.clj

```diff
@@ -31,39 +38,67 @@
-                  (when (and new-value (str/blank? (embed/embedding-secret-key)))
-                    (embed/embedding-secret-key! (crypto-random/hex 32)))
-                  (snowplow/track-event! ::snowplow/embed_share
-                                         {:event                      (if new-value
-                                                                        :embedding-enabled
-                                                                        :embedding-disabled)
-                                          :embedding-app-origin-set   (boolean (embedding-app-origin))
-                                          :number-embedded-questions  (t2/count :model/Card :enable_embedding true)
-                                          :number-embedded-dashboards (t2/count :model/Dashboard :enable_embedding true)}))))
+  :deprecated "0.51.0"
+  :setter     (make-embedding-toggle-setter :enable-embedding "embedding"))
 
-(defsetting enable-embedding-static
-  (deferred-tru "Allow admins to embed Metabase via static embedding?")
+(defsetting embedding-app-origin
+  (deferred-tru "Allow this origin to embed the full Metabase application.")
+  ;; This value is usually gated by [[enable-embedding]]
+  :feature    :embedding
+  :type       :string
+  :export?    false
+  :visibility :public
+  :audit      :getter)
+
+(defsetting enable-embedding-sdk
+  (deferred-tru "Allow admins to embed Metabase via the SDK?")
   :type       :boolean
   :default    false
   :visibility :authenticated
   :export?    false
   :audit      :getter
-  :setter     (fn [new-value]
-                (when (not= new-value (setting/get-value-of-type :boolean :enable-embedding-static))
-                  (setting/set-value-of-type! :boolean :enable-embedding-static new-value)
-                  (when (and new-value (str/blank? (embed/embedding-secret-key)))
-                    (embed/embedding-secret-key! (crypto-random/hex 32))))))
+  :setter     (make-embedding-toggle-setter :enable-embedding-sdk "sdk-embedding"))
 
-(defsetting enable-embedding-sdk
-  (deferred-tru "Allow admins to embed Metabase via the SDK?")
+(mu/defn- ignore-localhost :- :string
+  "Remove localhost:* or localhost:<port> from the list of origins."
+  [s :- [:maybe :string]]
+  (->> (str/split (or s "") #"\s+")
+       (remove #(re-matches #"localhost:(\*|\d+)" %))
+       distinct
+       (str/join " ")
+       str/trim))
+
+(mu/defn- add-localhost :- :string [s :- [:maybe :string]]
+  (->> s ignore-localhost (str "localhost:* ") str/trim))
+
+(defsetting embedding-app-origins-sdk
+  (deferred-tru "Allow this origin to embed Metabase SDK")
+  :type       :string
+  :export?    false
+  :visibility :public
+  :encryption :never
+  :audit      :getter
+  :getter    (fn embedding-app-origins-sdk-getter []
+               (when (enable-embedding-sdk)
+                 (add-localhost (setting/get-value-of-type :string :embedding-app-origins-sdk))))
```

Comment:
Not sure what the product requirements are, but should this match the setter and do a token check, and return just localhost if the feature isn't available?

Reply:
Yeah, I think I was doing too much in the setting itself. I made it so it matches the setter, and I've changed it's usages to check for `embed.settings/enable-embedding-sdk` instead.